go/adbc/driver/snowflake/connection.go

```diff
@@ -326,7 +326,7 @@ func (c *cnxn) getObjectsDbSchemas(ctx context.Context, depth adbc.ObjectDepth,
 				IF (counter > 0) THEN
 				  statement := statement || ' UNION ALL ';
 				END IF;
-				statement := statement || ' SELECT CATALOG_NAME, SCHEMA_NAME FROM ' || rec.database_name || '.INFORMATION_SCHEMA.SCHEMATA';
+				statement := statement || ' SELECT CATALOG_NAME, SCHEMA_NAME FROM "' || rec.database_name || '".INFORMATION_SCHEMA.SCHEMATA';
```

Comment:
Can database names contain a single quote?

Reply:
Not per https://docs.snowflake.com/en/sql-reference/identifiers-syntax#double-quoted-identifiers, but the quotes does make them case-sensitive.

Reply:
Sorry, I meant to say "double quote". That same link says "To use the double quote character inside a quoted identifier, use two quotes" and as an example it gives

create table "quote""andunquote"""

If a table name can contain a double quote then perhaps a database name can also?

Reply:
I was doing some testing with this. I think it's a bigger problem than what should be squeezed into this PR. I will file an issue and look at it separately.

Reply:
Filed https://github.com/apache/arrow-adbc/issues/1225